ext/pdo/pdo_stmt.c

```diff
@@ -688,13 +634,84 @@ static bool pdo_do_key_pair_fetch(pdo_stmt_t *stmt, enum pdo_fetch_orientation o
 	return true;
 }
 
+/* Previous the legacy zend_fcall_info_args_ex() function was used to set the
+ * constructor arguments. The particularity of this function is to it "wraps" arguments
+ * that need to be passed by-ref in a reference.
+ * As this API only supported positional args, we don't bother doing this for named arguments. */
+static HashTable* pdo_class_fetch_prepare_constructor_arguments(const zend_function *constructor, HashTable *ctor_args)
+{
+	zend_ulong index;
+	zend_string *arg_name = NULL;
+	zval *arg_value = NULL;
+	bool needs_duplication = false;
+	ZEND_HASH_FOREACH_KEY_VAL(ctor_args, index, arg_name, arg_value) {
+		/* Named param, means we can assume no positional arguments exist after */
+		if (arg_name) {
+			break;
+		}
+		if (!Z_ISREF_P(arg_value) && ARG_SHOULD_BE_SENT_BY_REF(constructor, index+1)) {
+			needs_duplication = true;
+			break;
+		}
+	} ZEND_HASH_FOREACH_END();
+	if (needs_duplication) {
+		HashTable *duplicated = zend_new_array(zend_hash_num_elements(ctor_args));
+		ZEND_HASH_FOREACH_KEY_VAL(ctor_args, index, arg_name, arg_value) {
```

Comment:
But the array is released afterwards, right? So why isn't the refcount of the values increased then?